chats/serializers.py

```diff
@@ -0,0 +1,64 @@
+from django.db.models import Q
+from rest_framework import serializers
+from chats.models import Chat, Message
+from companies.models import Companies, CompaniesAndUsersRelations
+
+
+class ParticipantSerializer(serializers.ModelSerializer):
+    brand = serializers.SerializerMethodField()
+
+    def get_brand(self, participant):
+        return participant.brand if participant else None
+
+    class Meta:
+        model = Companies
+        fields = ['company_id', 'brand']
+
+
+class MessageSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Message
+        fields = ['message_id', 'sender', 'recipient', 'content']
+        ordering = ['timestamp']
+
+    def validate(self, data):
+        sender_id = self.context['request'].user.id
+        sender_company_relations = CompaniesAndUsersRelations.objects.filter(user_id=sender_id)
+
+        if not sender_company_relations.exists():
+            raise serializers.ValidationError("You dont have companies in your list.")
+
+        sender_company_ids = [relation.company_id for relation in sender_company_relations]
+        sender_company_id = data.get('sender')
+        if sender_company_id not in sender_company_ids:
+            raise serializers.ValidationError("You are not allowed to send message")
+
+        return data
+
+    def create(self, validated_data):
+        participant_1 = validated_data["sender"]
+        participant_2 = validated_data["recipient"]
```

Comment:
You mean validate it it in case if such company exists? Or implement, that sender can not be the same as recipient?

Reply:
In my opinion, the logic of creation messages should be in views.py, because serializers made for validation. Single-responsibility principle. @mehalyna What do you think?

Reply:
agree, updating code, logic of creating messages ll be moved to views